src/main.cairo

```diff
@@ -180,6 +192,54 @@ func prolong_auction_on_end{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, ra
     return ()
 end
 
+func verify_auction_does_not_exist{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(auction_id):
+    alloc_locals
+
+    with_attr error_message(
+        "Auction {auction_id} already exists!"
+    ):
+        let (existing_auction) = auctions.read(auction_id)
+        assert existing_auction.seller = 0
+        assert existing_auction.asset_id = Uint256(0, 0)
+    end
+
+    return ()
+end
+
+func verify_address{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(address):
+    with_attr error_message(
+        "Invalid address {address}"
+    ):
+        assert_not_zero(address)
+    end
+
+    return ()
+end
+
+func verify_min_bid_increment{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        min_bid_increment : Uint256
+    ):
+    with_attr error_message(
+        "Invalid minimal value for bid, has to be >= 0"
+    ):
+        let (value) = uint256_lt(Uint256(1, 0), min_bid_increment)
+        assert value = 1
+    end
+
+    return ()
+end
+
+func verify_lifetime{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(lifetime):
+    with_attr error_message(
+        "Invalid lifetime, has to be >= 0"
```

Comment:
>

Reply:
If lifetime is 0, doesn't that mean the auction will end in the same block it was created? Should we allow for that?